### PR TITLE
Models and repositories

### PR DESCRIPTION
**Pull request by cookiecuttershark**
Added models and repositories, and updated item_reviews to add a user column.

### CODE REVIEW

File: src/main/java/com/launchacademy/giantleap/models/ItemReview.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+//    CREATE TABLE fashion_item_reviews (
+//            id SERIAL PRIMARY KEY,
+//            user_id INTEGER REFERENCES users(id),
+//    body_Type VARCHAR(255),
+//    style VARCHAR(255),
+//    item_size INTEGER ,
+//    quality INTEGER,
+//    comment VARCHAR(255),
+//    photo VARCHAR(255)
+//);
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/giantleap/models/ItemReview.java]: please remove commented out code prior to submitting PRs

File: src/main/java/com/launchacademy/giantleap/models/Style.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      generator = "style_generator")
+  @Column(name="id", nullable=false, unique=true)
+  private Integer id;
+  @NonNull
+  @Column(nullable = false)
+  private String name;
+
+  @ManyToMany(mappedBy = "styles")
+  private Set<FashionItem> fashionItems = new HashSet<FashionItem>();
+
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/giantleap/models/Style.java]: please remove extra line

File: src/main/java/com/launchacademy/giantleap/models/Style.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
+package com.launchacademy.giantleap.models;
+
+
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/giantleap/models/Style.java]: please remove extra line

File: src/main/java/com/launchacademy/giantleap/models/Brand.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
+package com.launchacademy.giantleap.models;
+
+
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/giantleap/models/Brand.java]: please remove extra line

File: src/main/java/com/launchacademy/giantleap/models/Budget.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
+package com.launchacademy.giantleap.models;
+
+
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/giantleap/models/Budget.java]: please remove extra line

File: src/main/java/com/launchacademy/giantleap/models/ClothingType.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
+package com.launchacademy.giantleap.models;
+import java.util.ArrayList;
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/giantleap/models/ClothingType.java]: please add a line between the package and the imports

File: src/main/java/com/launchacademy/giantleap/models/FashionItem.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  @Column(name="fabric_weight")
+  private Integer fabricWeight;
+
+  @Column(name="body_type")
+  private String bodyType;
+
+  @Column(name="item_size")
+  private Integer itemSize;
+
+  //many to many
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/giantleap/models/FashionItem.java]: please remove comments prior to submitting PRs

File: src/main/java/com/launchacademy/giantleap/models/ClothingType.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+@Setter
+@NoArgsConstructor
+
+public class ClothingType {
+  @Id
+  @SequenceGenerator(name= "clothingtype_generator", sequenceName = "clothingtype_id_seq", allocationSize = 1)
+  @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "clothingtype_generator")
+  @Column(name="id", nullable=false, unique=true)
+  private Integer id;
+  @NonNull
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/giantleap/models/ClothingType.java]: please add line between columns

File: src/main/java/com/launchacademy/giantleap/models/FashionItem.java [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+@Getter
+@Setter
+@NoArgsConstructor
+public class FashionItem {
+  @Id
+  @SequenceGenerator(name= "fashion_items_generator", sequenceName = "fashion_items_id_seq", allocationSize = 1)
+  @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "fashion_items_generator")
+  @Column(name="id", nullable=false, unique=true)
+  private Integer id;
+
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/giantleap/models/FashionItem.java]: currently these are all optional, from the model perspective. Is that intentional? 
If they are null false in the DB, ensure the model matches.

File: src/main/java/com/launchacademy/giantleap/models/Brand.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+public class Brand {
+  @Id
+  @SequenceGenerator(name= "brand_generator",
+      sequenceName = "brand_id_seq", allocationSize = 1)
+  @GeneratedValue(strategy = GenerationType.SEQUENCE,
+      generator = "brand_generator")
+  @Column(name="id", nullable=false, unique=true)
+  private Integer id;
+  @NonNull
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/giantleap/models/Brand.java]: please add line between columns

File: src/main/resources/db/migration/V3__create_review_table.sql [hunk 11]
```diff
@@ -1,6 +1,6 @@
 CREATE TABLE fashion_item_reviews (
```
Comment (prophetsfall) [inline src/main/resources/db/migration/V3__create_review_table.sql]: per the article the reviews require a rating.

File: src/main/java/com/launchacademy/giantleap/models/ItemReview.java [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+import org.hibernate.validator.constraints.URL;
+
+import javax.persistence.*;
+
+@Entity
+@Table(name="fashion_item_reviews")
+@Getter
+@Setter
+@NoArgsConstructor
+public class ItemReview  {
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/giantleap/models/ItemReview.java]: if you're calling your class FashionItem I would suggest naming this FashionItemReview for clarity's sake.

File: src/main/java/com/launchacademy/giantleap/models/ItemReview.java [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+
+    @Column(name="quality")
+    private Integer quality;
+
+    @Column(name="comment")
+    private String comment;
+
+    @URL
+    @Column
+    private String photo;
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/giantleap/models/ItemReview.java]: the review requires a rating, please add.

File: src/main/java/com/launchacademy/giantleap/models/FashionItem.java (hunk 9 above)
Comment (cookiecuttershark) [reply src/main/java/com/launchacademy/giantleap/models/FashionItem.java]: Yes, it is intentional.

File: src/main/java/com/launchacademy/giantleap/models/Budget.java [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+@Setter
+@NoArgsConstructor
+
+public class Budget {
+  @Id
+  @SequenceGenerator(name = "budget_generator", sequenceName = "budget_id_seq", allocationSize = 1)
+  @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "budget_generator")
+  @Column(name = "id", nullable = false, unique = true)
+  private Integer id;
+  @NonNull
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/giantleap/models/Budget.java]: I missed this one. I'm going to merge but please add a line here for next time.

**Pull request merged**